### PR TITLE
Carson/v4

### PR DESCRIPTION
**Pull request by carsonfarmer**
Hey @joewagner and @awmuncy can you check and make sure we are prepared to release v4 with this PR?

### CODE REVIEW

**Comment by joewagner** [comment]
@carsonfarmer This package, `@tableland/local`, and `@tableland/sdk` all depend on each other. This is ok, but we will need to make those updates in the `package.json`s in a somewhat synchronized way.  Do we want to update the `@tableland/local` and `@tableland/sdk` versions in this PR?

File: package.json [hunk 1]
```diff
@@ -58,24 +58,24 @@
     "@nomicfoundation/hardhat-chai-matchers": "^1.0.4",
     "@nomiclabs/hardhat-ethers": "^2.0.6",
     "@nomiclabs/hardhat-etherscan": "^3.1.0",
-    "@openzeppelin/contracts": "^4.7.3",
-    "@openzeppelin/contracts-upgradeable": "^4.7.3",
+    "@openzeppelin/contracts": "4.7",
+    "@openzeppelin/contracts-upgradeable": "4.7",
     "@openzeppelin/hardhat-upgrades": "^1.19.0",
     "@tableland/local": "^0.0.3",
     "@tableland/sdk": "^3.1.4",
```
Comment (joewagner) [inline package.json:65]: Can we change these lines to: 
```
"@tableland/local": "^1.0.0-pre.1",
"@tableland/sdk": "^4.0.0-pre.2",
```
?

Comment (carsonfarmer) [reply package.json:65]: yes of course, good point!

Comment (carsonfarmer) [reply package.json:65]: oof, circular deps are tricky. at least just dev deps on this end.

Comment (carsonfarmer) [reply package.json:65]: oh wait, the more recent update for @tabeland/local on npm is 0.0.5-pre.2, which is a month old. do we wanna publish the 1.0.0-pre.1 first and then do the update here?

Comment (carsonfarmer) [reply package.json:65]: also, i'll do another pre-release on the sdk so we can use that

**Review by joewagner — DISMISSED** [review]
All of this looks good to go to main.  In a different PR we might want to setup the same new publishing flow we are using in the sdk et al.

**Comment by carsonfarmer** [comment]
Ok @joewagner and @awmuncy , with the updated local-tableland, the tests are passing again. Let's get this one merged and released, and then update the downstream deps before EOD to make sure we're 100% for next week's releases.

**Review by awmuncy — APPROVED** [review]

**Pull request merged**